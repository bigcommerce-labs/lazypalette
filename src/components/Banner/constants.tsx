import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
