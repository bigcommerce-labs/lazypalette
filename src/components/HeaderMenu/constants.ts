export const enum HeaderMenuLinks {
    CONTROL_PANEL = 'manage',
}

export const enum HeaderMenuMessages {
    GO_BACK = 'Go back to Control Panel',
}
