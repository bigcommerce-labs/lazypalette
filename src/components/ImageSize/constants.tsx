export const enum Messages {
    Dimension = 'Specify a maximum image size for desktop display. We\'ll automatically ' +
        'scale down the image for smaller devices.',
    Lock = 'Constrain Proportions',
}

export const ImageDimensionPattern = (/^[\d]+x[\d]+$/);
