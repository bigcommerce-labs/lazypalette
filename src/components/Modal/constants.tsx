export const enum Messages {
    Reset = 'This theme has unpublished changes. Do you want to proceed?',
    Variation = 'This variation has unpublished changes. Do you want to proceed?',
}
