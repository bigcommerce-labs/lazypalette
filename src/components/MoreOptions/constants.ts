export const RestoreOriginalText = {
    action: 'Restore',
    body: 'All changes to this theme will be restored back to its default settings. ' +
    'This will not publish changes to your live store.',
    title: 'Restore original theme style',
};
