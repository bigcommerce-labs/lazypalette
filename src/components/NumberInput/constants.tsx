export const enum Status {
    Invalid = 'Invalid',
    Undefined = 'Undefined',
}

export const enum Notes {
    Required = 'Required*',
}

export const enum Sign {
    Minus = '-',
    Plus = '+',
}
