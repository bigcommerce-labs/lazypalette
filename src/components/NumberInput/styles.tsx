import { Theme } from 'pattern-lab';
import styled from 'styled-components';

import { Sign, Status } from './constants';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
`;

export const Label = styled.label`
    padding: 0 0 .5rem;
    color:  ${({ theme }) => theme.colors.primaryText};
    line-height: 1rem;
`;

interface BorderRadiusProps {
    value: string;
    theme: Theme;
}

const getBorderRadius = ({value, theme}: BorderRadiusProps) => {
    const defaultRadius = 'border-radius: 2px';

    if (value === Sign.Plus) {
        return `
            ${defaultRadius};
            border-top-left-radius: 0px;
            border-bottom-left-radius: 0px;
        `;
    } else if (value === Sign.Minus) {
        return `
            ${defaultRadius};
            border-top-right-radius: 0px;
            border-bottom-right-radius: 0px;
        `;
    }

    return `
        ${defaultRadius};
    `;
};

interface ButtonProps {
    value: string;
}

export const Button = styled.input.attrs<ButtonProps>({})`
    height: 2rem;
    border: 1px solid ${({ theme }) => theme.colors.stroke};
    ${props => getBorderRadius(props)};
    padding: 0 0 .25rem;
    font-size: 1.5rem;
    cursor: pointer;
    background-color: ${({ theme }) => theme.colors.empty};
    color: ${({ theme }) => theme.colors.primary};
    width: 2.5rem;

    :hover {
        border-color: ${({ theme }) => theme.colors.primary};
    }

    :focus {
        outline: none !important;
    }

    :disabled, :disabled:hover {
        background-color: ${({ theme }) => theme.colors.empty};
        border: 1px solid ${({ theme }) => theme.colors.stroke};
        opacity: 0.5;
        cursor: not-allowed;
    }
`;

interface BorderProps {
    status?: string;
    theme: Theme;
}

const getBorder = ({status, theme}: BorderProps) => {
    if (status === Status.Invalid) {
        return `
            border: 1px solid ${theme.colors.error};
        `;
    }

    return `
        border: 1px solid ${theme.colors.stroke};
        border-left: none;
        border-right: none;
    `;
};

interface InputProps {
    status?: string;
}

export const Input = styled.input.attrs<InputProps>({})`
    cursor: text;
    background-color: ${({ theme }) => theme.colors.empty};
    ${props => getBorder(props)};
    box-sizing: border-box;
    height: 2rem;
    padding: .25rem;
    font-size: 0.75rem;
    text-align: center;
    width: 2.5rem;

    :focus {
        border-color: ${({ theme }) => theme.colors.stroke};
        outline: none;
    }

    :disabled {
        opacity: 0.5;
        cursor: not-allowed;
    }

    ::-webkit-inner-spin-button, ::-webkit-outer-spin-button {
        -webkit-appearance: none;
        margin: 0;
    }
`;

export const InputBox = styled.div`
    display: flex;
`;

interface SmallProps {
    required?: boolean;
}

export const Small = styled.small.attrs<SmallProps>({})`
    ${({ required, theme }) => `
        color: ${required ? theme.colors.error : 'initial'};
        font-size: ${theme.typography.fontSize.smallest};
        padding-top: .25rem;
    `}
`;
