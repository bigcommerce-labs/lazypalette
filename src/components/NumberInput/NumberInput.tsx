import React, {
    ChangeEvent,
    ChangeEventHandler,
    FocusEvent,
    FocusEventHandler,
    MouseEvent,
    PureComponent,
} from 'react';
import uuid from 'uuid';

import { Notes, Sign, Status } from './constants';
import { Button, Container, Input, InputBox, Label, Small } from './styles';

interface NumberInputProps extends Partial <{
    disabled: boolean;
    inputId: string;
    label: string;
    max: number;
    min: number;
    required: boolean;
    value: string;
    onBlur: FocusEventHandler<HTMLInputElement>;
    onChange: ChangeEventHandler<HTMLInputElement>;
}> {}

interface NumberInputState {
    inputId: string;
    status: Status;
    value: string;
}

class NumberInput extends PureComponent<NumberInputProps, NumberInputState> {
    readonly state: NumberInputState = {
        inputId: this.props.inputId || uuid(),
        status: Status.Undefined,
        value: this.props.value || '0',
    };

    componentDidUpdate(prevProps: NumberInputProps) {
        const { value } = this.props;
        if (value && prevProps.value !== value) {
            this.setState({ value });
        }
    }

    handleBlur = (event: FocusEvent<HTMLInputElement>) => {
        const newStatus = this.isInvalid() ? Status.Invalid : Status.Undefined;
        if (this.state.status !== newStatus) {
            this.setState({ status: newStatus });
        }

        if (this.props.onBlur) {
            this.props.onBlur(event);
        }
    };

    handleButtonClick = (event: MouseEvent<HTMLInputElement>) => {
        const { max, min } = this.props;
        const { value: button } = event.target as HTMLInputElement;
        const { value: prevValue } = this.state;
        const numValue = Number(prevValue);
        let newValue = numValue;

        if (button === Sign.Plus) {
            newValue = max && numValue >= max ? max : numValue + 1;
        } else if (button === Sign.Minus) {
            newValue = min && numValue <= min ? min : numValue - 1;
        }

        if (`${newValue}` !== prevValue) {
            const newEvent = {
                ...event,
                target: { ...event.target as HTMLInputElement, value: `${newValue}` },
            };
            this.handleChange(newEvent);
        }
    };

    handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        this.setState({ status: Status.Undefined, value: event.target.value });

        if (this.props.onChange) {
            this.props.onChange(event);
        }
    };

    isInvalid = () => !this.state.value && this.props.required;

    render() {
        const { inputId, status, value } = this.state;
        const note = this.isInvalid() ? Notes.Required : '';
        const {
            disabled,
            label,
            max,
            min,
            required,
        } = this.props;
        const numValue = Number(value);
        const isMinDisabled = min ? numValue <= min : numValue < 1;
        const isMaxDisabled = max ? numValue >= max : false;

        return (
            <Container>
                <Label htmlFor={inputId}>
                    {label}
                </Label>
                <InputBox>
                    <Button
                        onClick={this.handleButtonClick}
                        disabled={isMinDisabled}
                        type="button"
                        value={Sign.Minus}
                    />
                    <Input
                        disabled={disabled}
                        id={inputId}
                        max={max}
                        min={min}
                        onBlur={this.handleBlur}
                        onChange={this.handleChange}
                        required={required}
                        status={status}
                        type="number"
                        value={value}
                    />
                    <Button
                        onClick={this.handleButtonClick}
                        disabled={isMaxDisabled}
                        type="button"
                        value={Sign.Plus}
                    />
                </InputBox>
                {note &&
                    <Small required={required}>
                        {note}
                    </Small>
                }
            </Container>
        );
    }
}

export default NumberInput;
