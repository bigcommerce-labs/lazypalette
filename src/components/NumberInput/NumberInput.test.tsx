import { shallow } from 'enzyme';
import React from 'react';

import { Sign, Status } from './constants';
import NumberInput from './NumberInput';

const numId = 'meow123';

describe('NumberInput', () => {
    it('renders', () => {
        const numberInput = shallow(
            <NumberInput
                disabled={false}
                inputId={numId}
                label="FooBar"
                value="5"
            />
        );
        expect(numberInput).toMatchSnapshot();
    });

    it('should render a default state', () => {
        const numberInput = shallow(
            <NumberInput
                inputId={numId}
                label="Max width"
                value="12"
            />
        );

        numberInput.find(`[id="${numId}"]`).simulate('blur', {});

        expect(numberInput).toMatchSnapshot();
    });

    it('should render an invalid state for required fields', () => {
        const numberInput = shallow(
            <NumberInput
                inputId={numId}
                label="Min height"
                required={true}
            />
        );

        numberInput.find(`[id="${numId}"]`).simulate('blur', {});

        expect(numberInput).toMatchSnapshot();
    });

    describe('when the user changes the input', () => {
        it('should call the onChange handler', () => {
            const mockHandler = jest.fn();
            const event = {
                target: { value: 'happyCat' },
            };
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="depth"
                    onChange={mockHandler}
                />
            );
            numberInput.find(`[id="${numId}"]`).simulate('change', event);
            expect(mockHandler).toBeCalledWith(event);
        });
    });

    describe('when the user clicks off the input area', () => {
        it('should call the onBlur handler', () => {
            const mockHandler = jest.fn();
            const event = {
                target: { value: 'happyCat' },
            };
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="volume"
                    onBlur={mockHandler}
                    required={true}
                />
            );
            numberInput.find(`[id="${numId}"]`).simulate('blur', event);
            expect(mockHandler).toBeCalledWith(event);
        });

        it('should cause an invaild status if the input is cleared', () => {
            const mockOnBlur = jest.fn();
            const event = {
                target: { value: '' },
            };
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="radius"
                    onBlur={mockOnBlur}
                    required={true}
                />
            );
            numberInput.find(`[id="${numId}"]`).simulate('change', event);
            numberInput.find(`[id="${numId}"]`).simulate('blur', event);
            expect(numberInput.state('status')).toBe(Status.Invalid);
        });
    });

    describe('when the user clicks the plus button', () => {
        const clickEvent = {
            target: { value: '+' },
        };

        it('should call the onChange handler with value + 1', () => {
            const mockHandler = jest.fn();
            const initialValue = '10';
            const changeEvent = {
                target: { value: `${+initialValue + 1}` },
            };
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="depth"
                    onChange={mockHandler}
                    value={initialValue}
                />
            );

            expect(numberInput.state().value).toEqual(initialValue);

            const addButton = numberInput.find({ value: Sign.Plus });
            addButton.simulate('click', clickEvent);

            expect(mockHandler).toBeCalledWith(changeEvent);
            expect(numberInput.state().value).toEqual('11');
        });

        describe('when a max prop is passed in', () => {
            const mockHandler = jest.fn();
            const initialValue = '6';
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="depth"
                    onChange={mockHandler}
                    max={6}
                    value={initialValue}
                />
            );

            it('should not increment the value past the max', () => {
                expect(numberInput.state().value).toEqual(initialValue);

                const addButton = numberInput.find({ value: Sign.Plus });
                addButton.simulate('click', clickEvent);

                expect(mockHandler).toHaveBeenCalledTimes(0);
                expect(numberInput.state().value).toEqual(initialValue);
            });
        });
    });

    describe('when the user clicks the minus button', () => {
        const clickEvent = {
            target: { value: '-' },
        };

        it('should call the onChange handler with value - 1', () => {
            const mockHandler = jest.fn();
            const initialValue = '7';
            const changeEvent = {
                target: { value: `${+initialValue - 1}` },
            };
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="depth"
                    onChange={mockHandler}
                    value={initialValue}
                />
            );

            expect(numberInput.state().value).toEqual(initialValue);

            const minusButton = numberInput.find({ value: Sign.Minus });
            minusButton.simulate('click', clickEvent);

            expect(mockHandler).toBeCalledWith(changeEvent);
            expect(numberInput.state().value).toEqual('6');
        });

        describe('when a min prop is passed in', () => {
            const mockHandler = jest.fn();
            const initialValue = '1';
            const numberInput = shallow(
                <NumberInput
                    inputId={numId}
                    label="depth"
                    onChange={mockHandler}
                    min={1}
                    value={initialValue}
                />
            );

            it('should not decrement the value past the min', () => {
                expect(numberInput.state().value).toEqual(initialValue);

                const minusButton = numberInput.find({ value: Sign.Minus });
                minusButton.simulate('click', clickEvent);

                expect(mockHandler).toHaveBeenCalledTimes(0);
                expect(numberInput.state().value).toEqual(initialValue);
            });
        });
    });
});
