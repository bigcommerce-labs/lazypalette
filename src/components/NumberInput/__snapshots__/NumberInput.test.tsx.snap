// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders 1`] = `
<styled.div>
  <styled.label
    htmlFor="meow123"
  >
    FooBar
  </styled.label>
  <styled.div>
    <styled.input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="-"
    />
    <styled.input
      disabled={false}
      id="meow123"
      onBlur={[Function]}
      onChange={[Function]}
      status="Undefined"
      type="number"
      value="5"
    />
    <styled.input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="+"
    />
  </styled.div>
</styled.div>
`;

exports[`NumberInput should render a default state 1`] = `
<styled.div>
  <styled.label
    htmlFor="meow123"
  >
    Max width
  </styled.label>
  <styled.div>
    <styled.input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="-"
    />
    <styled.input
      id="meow123"
      onBlur={[Function]}
      onChange={[Function]}
      status="Undefined"
      type="number"
      value="12"
    />
    <styled.input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="+"
    />
  </styled.div>
</styled.div>
`;

exports[`NumberInput should render an invalid state for required fields 1`] = `
<styled.div>
  <styled.label
    htmlFor="meow123"
  >
    Min height
  </styled.label>
  <styled.div>
    <styled.input
      disabled={true}
      onClick={[Function]}
      type="button"
      value="-"
    />
    <styled.input
      id="meow123"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      status="Undefined"
      type="number"
      value="0"
    />
    <styled.input
      disabled={false}
      onClick={[Function]}
      type="button"
      value="+"
    />
  </styled.div>
</styled.div>
`;
